VERSIONING.md

```diff
@@ -0,0 +1,104 @@
+# Versioning details
+
+This follows the [OpenTelemetry versioning proposal](https://github.com/open-telemetry/oteps/pull/143/files)
+
+OpenTelemetry .NET follows [SemVer V2](https://semver.org/spec/v2.0.0.html)
+guidelines. This means that, for any packages released from this repo, all
+public APIs will remain backward compatible, unless a major version bump occurs.
+This applies to the API, SDK, as well as Exporters, Instrumentation etc. shipped
+from this repo.
+
+For example, users can take a dependency on 1.0.0 version of any package, with
+the assurance that all future releases until 2.0.0 will be backward compatible.
+Any method/function which are planned to be removed in 2.0, will be marked
+[Obsolete](https://docs.microsoft.com/dotnet/api/system.obsoleteattribute)
+first.
+
+## Pre-releases
+
+Pre-release packages are denoted by appending identifiers such as
+-Alpha, -Beta, -RC
+etc. There are no API guarantees in pre-releases. In general, an RC pre-release
+is more stable than a Beta release, which is more stable than an Alpha release.
+
+### Public API change detection
+
+For convenience, every project in this repo uses
+[PublicApiAnalyzers](https://github.com/dotnet/roslyn-analyzers/tree/master/src/PublicApiAnalyzers)
+and lists public API in the directory "publicAPI". Any changes to public API,
+without corresponding changes here will result in build breaks - this helps
+catch any unintended changes to public API from being shipped accidentally. This
+also helps reviewers quickly understand if a given PR is making public API
+changes. For example,
+[this](https://github.com/open-telemetry/opentelemetry-dotnet/tree/master/src/OpenTelemetry.Instrumentation.AspNetCore/.publicApi)
+shows the public per target framework for the
+`OpenTelemetry.Instrumentation.AspNetCore` package.
+
+Since no stable version has been released so far, every API is listed in the
+"Unshipped.txt" file. Once 1.0.0 is shipped, it'll be moved to "Shipped.txt"
+file.
+
+## Packaging
+
+OpenTelemetry is structured around signals like Traces, Metrics, Logs, Baggage
+etc. OpenTelemetry .NET does not ship separate packages per signal. There is a
+single package which includes all the signals. i.e `OpenTelemetry.API` package
+will consist of API components from *all* the signals. `OpenTelemetry` package
+will consist of SDK components from *all* the signals. Instrumentations also
```

Comment:
Something interesting that was requested in Java, was the ability to offer a MicroMeter-based Metrics implementation, which would mean that each SDK signal portion would exist in its own package.

Reply:
I actually realized that there is some more clarifications required here. For SDK, its true that we have single package for traces and metrics. For API, its somewhat complex, as most of Tracing API is provided by `System.Diagnostics.DiagnosticSource` package, Context/Propagation is provided by `OpenTelemetry.API`. We do not know yet whether .NET team will use the `System.Diagnostics.DiagnosticSource` package itself for metrics, or a brand new package...